dxr/app.py

```diff
@@ -170,6 +170,20 @@ def _search_json(query, tree, query_text, is_case_sensitive, offset, limit, conf
             return jsonify({'redirect': url_for('.browse', _anchor=line, **params)})
     try:
         count_and_results = query.results(offset, limit)
+        # If we're asked to redirect and there's a single result, redirect to the result.
+        if (request.values.get('redirect') == 'true' and
+            count_and_results['result_count'] == 1):
+            _, path, line, _ = next(count_and_results['results'])
+            line = line[0][0] if line else None
```

Comment:
> It's worth considering whether it's desirable to redirect the user to a found FILE-domain result

My first reaction when I considered this point was to not redirect a FILE-domain result.  On the other hand, I think Gijs makes a pretty good argument in https://bugzilla.mozilla.org/show_bug.cgi?id=1202603  At some point it was decided that it was a good idea to redirect to a FILE-domain result on direct search, presumably on the theory that you were searching on a path name.  If you explicitly specify that you're searching for a path name by using a `path:` filter, why should that change?  Also if you don't want the redirect you can just not hit return (though I have to say that behavior was confusing to me pretty much right up to the point where I started reading this code).